spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/repository/query/ReactiveNeo4jQueryMethod.java

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.neo4j.springframework.data.repository.query;
+
+import static org.springframework.data.repository.util.ClassUtils.*;
+
+import java.lang.reflect.Method;
+
+import org.springframework.dao.InvalidDataAccessApiUsageException;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.projection.ProjectionFactory;
+import org.springframework.data.repository.core.RepositoryMetadata;
+import org.springframework.data.repository.util.ReactiveWrappers;
+import org.springframework.data.util.ClassTypeInformation;
+import org.springframework.data.util.TypeInformation;
+import org.springframework.util.ClassUtils;
+
+/**
+ * This is unfortunately a little bit of a hack to provide the information that the returned types by this query
+ * are always considered as stream.
+ * We try to either separate imperative and reactive concerns but due to type compatibility we extend the
+ * {@link Neo4jQueryMethod} here instead of creating a complete new reactive focused logical branch.
+ * It would only contain duplications of several classes.
+ *
+ * @author Gerrit Meier
+ * @since 1.0
+ */
+final class ReactiveNeo4jQueryMethod extends Neo4jQueryMethod {
+
+	private static final ClassTypeInformation<Page> PAGE_TYPE = ClassTypeInformation.from(Page.class);
+	private static final ClassTypeInformation<Slice> SLICE_TYPE = ClassTypeInformation.from(Slice.class);
+
+	/**
+	 * Creates a new {@link ReactiveNeo4jQueryMethod} from the given parameters.
+	 *
+	 * @param method   must not be {@literal null}.
+	 * @param metadata must not be {@literal null}.
+	 * @param factory  must not be {@literal null}.
+	 */
+	ReactiveNeo4jQueryMethod(Method method, RepositoryMetadata metadata, ProjectionFactory factory) {
+		super(method, metadata, factory);
+
+		if (hasParameterOfType(method, Pageable.class)) {
+
+			TypeInformation<?> returnType = ClassTypeInformation.fromReturnTypeOf(method);
+
+			boolean multiWrapper = ReactiveWrappers.isMultiValueType(returnType.getType());
+			boolean singleWrapperWithWrappedPageableResult = ReactiveWrappers.isSingleValueType(returnType.getType())
+				&& (PAGE_TYPE.isAssignableFrom(returnType.getRequiredComponentType())
+				|| SLICE_TYPE.isAssignableFrom(returnType.getRequiredComponentType()));
+
+			if (singleWrapperWithWrappedPageableResult) {
+				throw new InvalidDataAccessApiUsageException(
+					String.format("'%s.%s' must not use sliced or paged execution. Please use Flux.buffer(size, skip).",
+						ClassUtils.getShortName(method.getDeclaringClass()), method.getName()));
+			}
+
+			if (!multiWrapper) {
+				throw new IllegalStateException(String.format(
+					"Method has to use a either multi-item reactive wrapper return type or a wrapped Page/Slice type. "
```

Comment:
> It will not support using Page or Slice as the return type.

I don't know if I understood correctly, didn't you say that Page/Slice won't be supported? Then why the error message says that it can be a wrapped Page/Slice e.g. Flux<Page<Person>> ?